### PR TITLE
RSDK-6700 Bugfix for course correction

### PR DESCRIPTION
**Pull request by biotinker**
It was noted that course-correcting bases could sometimes veer wildly off the path.

It was determined that the goal poses being course corrected for could be incorrect in the case where their parent arc was one which had already been course-corrected.

### CODE REVIEW

File: components/base/kinematicbase/execution.go [hunk 1]
```diff
@@ -130,6 +130,16 @@ func (ptgk *ptgBaseKinematics) GoToInputs(ctx context.Context, inputSteps ...[]r
 		// - move until we think we have finished the arc, then move on to the next step
 		// - update our CurrentInputs tracking where we are through the arc
 		// - Check where we are relative to where we think we are, and tweak velocities accordingly
+
+		// Check if this arc is shorter than our typical check time; if so just run that and do not course correct.
+		if step.durationSeconds < updateStepSeconds {
+			utils.SelectContextOrWait(ctx, (time.Duration(step.durationSeconds*1000) * time.Millisecond))
```
Comment (raybjork) [inline components/base/kinematicbase/execution.go:136]: Can you not just write `step.durationSeconds` here?  The units on a `Duration` shouldnt matter IIRC

File: components/base/kinematicbase/execution.go [hunk 2]
```diff
@@ -513,9 +524,23 @@ func (ptgk *ptgBaseKinematics) makeCourseCorrectionGoals(
...
+			}
+			arcPose, err := ptgk.Kinematics().Transform(arcTrajInputs)
+			if err != nil {
+				return []courseCorrectionGoal{}
+			}
+
 			goalPose := spatialmath.PoseBetween(
 				currPose,
-				spatialmath.Compose(steps[i].arcSegment.StartPosition, steps[i].subTraj[goalTrajPtIdx].Pose),
+				spatialmath.Compose(steps[i].arcSegment.StartPosition, arcPose),
```
Comment (raybjork) [inline components/base/kinematicbase/execution.go:543]: Noticing a proliferation of code that composes these two Poses. Am I right in assuming they might simplify with @nfranczak 's work on "execution frames"

File: components/base/kinematicbase/execution.go (hunk 1 above)
Comment (biotinker) [reply components/base/kinematicbase/execution.go:136]: The units on a duration absolutely matter. A duration is an `int`, so attempting to pass a float e.g. 0.4 seconds will be truncated to zero.

File: components/base/kinematicbase/execution.go (hunk 2 above)
Comment (biotinker) [reply components/base/kinematicbase/execution.go:543]: Probably not super related to the execution frame work.

We're requesting a relative transform along an arc (arcPose) and then determining where that pose will place us in absolute terms relative to other poses we care about (arcSegment.StartPosition). Whether the absolute pose onto which we are composing comes from a segment, an execution frame, or something else, will all have this same paradigm.

**Review by raybjork — APPROVED** [review]

**Pull request merged**